### ISSUE TITLE
dotnet 5.0 traces have large percentage of unresolved modules in CPU stacks view

### ISSUE THREAD

**Issue by kayle** [microsoft/perfview#1326]
I'm seeing a huge percentage of exclusive samples that are not resolved to a module:

![image](https://user-images.githubusercontent.com/1664776/98850576-fb632980-2422-11eb-8eab-370b0741a0ea.png)

I'm trying to profile the grpc client/server benchmark on windows with .net 5. Is this a known issue on .net 5, or something specific to this scenario? 

Trace is uploaded here: https://github.com/kayle/perf-investigations/blob/main/grpc-client-dotnet5.etl

**Comment by kayle** [comment]
After attaching the debugger, I think CodeAddressesInModule is 0 for most of the dotnet dlls in the trace. Are there some extra events I need to be collecting that I don't have?

**Comment by kayle** [comment]
Oops, the first trace didn't have ClrEvents on which explains the widespread missing symbols. Same problem, but less severe with the following trace:

![image](https://user-images.githubusercontent.com/1664776/98859813-bd6d0200-2430-11eb-8b48-bf2b93d97601.png)

[PerfViewData.zip](https://github.com/microsoft/perfview/files/5526515/PerfViewData.zip)

**Comment by brianrob** [comment]
@kayle, this is most common when either rundown doesn't happen (e.g. you don't get events that describe JIT events), or the trace wasn't merged and was either copied off box or one or more binaries changed.  I don't see a log with the trace - do you happen to have it handy?  I'm curious if merge succeeded.

**Comment by kayle** [comment]
Sure, here's the log/symbols for the rest of the zip file (was too big to upload as an attachment before). I see errors that it couldn't find crossgen.exe. Maybe perfview could download it at runtime (https://github.com/dotnet/installer/issues/3438)?

[PerfViewData.log.txt](https://github.com/microsoft/perfview/files/5527342/PerfViewData.log.txt)

[PerfViewData.etl.zip](https://github.com/microsoft/perfview/files/5527339/PerfViewData.etl.zip)

**Comment by brianrob** [comment]
Looks like merge succeeded.  The crossgen errors should not be an issue - in most cases, you don't need to use crossgen anymore - this is something that was required in previous version, but our diagnostics story has gotten better.

I don't see anything obviously wrong at this point.  Does this repro if you set the verbosity to verbose instead of informational?  Also, does it repro if you don't specify a focus process?

**Comment by kayle** [comment]
Yes, it repros with -ClrEventLevel:Verbose and without a focus process. 

The repro is not difficult:
1. git clone https://github.com/grpc/grpc-dotnet
2. start server with dotnet run -c Release -p .\perf\benchmarkapps\GrpcAspNetCoreServer\ --protocol h2c
3. start client with dotnet run -c Release -p .\perf\benchmarkapps\GrpcClient\ -- -u http://localhost:5000 -c 10 --streams 50 -s unary -p h2c --grpcClientType grpcnetclient

I've tried collecting traces with the GUI or commands similar to PerfViewCollect.exe collect -TplEvents:None  -ClrEventLevel:Informational -ClrEvents:Default-Threading-ThreadTransfer -BufferSize:1024 -FocusProcess:GrpcClient.exe

The extra flags are to try to avoid some of these common events which affect overall CPU utilization:
- System.Diagnostics.Eventing.FrameworkEventSource/ThreadPoolDequeueWork
- Microsoft-Diagnostics-DiagnosticSource/Activity1Start/Start
- Microsoft-Windows-DotNETRuntime/ThreadPoolWorkerThread/Wait
- Microsoft-Windows-DotNETRuntime/ThreadPool/IODequeue

**Comment by kayle** [comment]
@brianrob After setting fold% to 0, it looks like the ?!? samples could be inlined calls or some other runtime overhead:

For example, this trivial IsCompleted method has 16 inclusive samples that call ?!?
![image](https://user-images.githubusercontent.com/1664776/99576548-2233d980-29a8-11eb-8f96-09c5c15827de.png)


```
                   /// <summary>Gets whether the <see cref="ConfiguredValueTaskAwaitable{TResult}"/> has completed.</summary>
                   public bool IsCompleted
                   {
                       [MethodImpl(MethodImplOptions.AggressiveInlining)]
 16.0 |                get => _value.IsCompleted;
                   }
```

and this method has 13 inclusive samples that call into ?!?

![image](https://user-images.githubusercontent.com/1664776/99577014-b9009600-29a8-11eb-90ef-197494a5c630.png)

```
 13.0 |        public static StringComparer OrdinalIgnoreCase => OrdinalIgnoreCaseComparer.Instance;
```

Why would perfview not map these samples to the correct module? Is more context needed for inlined samples?
